design/ceph/osd-migration.md

```diff
@@ -1,32 +1,20 @@
-# OSD Backend Store Migration
+# OSD Migration
 
 ## Summary
 
-- Configure the default OSD backend via the `cephCluster` CR so that new OSDs
-  are built with it.
-- Existing OSDs should be able to migrate to use a new backend store configured
-  via `cephCluster` CR. The migration process should include destroying the
-  existing OSDs one by one, wiping the drive, then recreating a new OSD on
-  that drive with the same ID.
+- Automatically replace all the existing OSDs
 
 ## Motivation:
 
-- Ceph uses BlueStore as a special-purpose storage backend designed specifically
-  for Ceph OSD workloads. The upstream Ceph team is working on improving
-  BlueStore in ways that will significantly improve performance, especially
-  in disaster recovery scenarios. This new OSD backend is known as
-  `bluestore-rdr`. In the future, other backends will also need to be supported,
-  such as `seastore` from the Crimson effort.
-- To employ a new torage backend, the existing OSD needs to be destroyed, drives
-  wiped, and new OSD created using the specified storage backend.
+- Certain scenarios like enabling encryption on existing OSDs or changing the OSD
+  backend store require the users to manually purge the OSDs and then create new ones.
+  Rook should be able to migrate the OSDs automatically without any user intervention.
 
 ## Goals
 
-- Newly provisioned clusters should be able to use the specific storage backend
-  when creating OSDs.
-- Existing clusters should be able to migrate existing OSDs to a specific
-  backend, without downtime and without risking data loss.  Note that this
-  necessarily entails redeploying and backfilling each OSD in turn.
+- Support automatic OSD migration for following scenarios:
+    - Change OSD backing store
+    - Enable OSD encryption as a day-2 operation
```

Comment:
What about disabling? That seems possible, but perhaps nobody would actually want to do it in reality.

Reply:
We can support both enabling and disabling.